llvm/include/llvm/BinaryFormat/DXContainer.h

```diff
@@ -546,15 +560,47 @@ struct ProgramSignatureElement {
 static_assert(sizeof(ProgramSignatureElement) == 32,
               "ProgramSignatureElement is misaligned");
 
-struct RootSignatureValidations {
+struct RootConstants {
+  uint32_t Register;
+  uint32_t Space;
+  uint32_t NumOfConstants;
```

Comment:
```suggestion
  uint32_t ShaderRegister;
  uint32_t RegisterSpace;
  uint32_t Num32BitValues;
```

We might as well match the names in the d3d structures.

Reply:
We are following the names established in the [binary representation documentation](https://github.com/llvm/llvm-project/blob/main/llvm/docs/DirectX/DXContainer.rst#root-signature-rts0-part), this was requested in a previous comment as well

Reply:
Ok, well that document should be fixed then.  Note that the naming style here doesn't match that for other fields later on in DXContainer.rst - eg "NumOfConstants" vs "NumDescriptors".  Or "Register" vs "BaseShaderRegister".

We should just match the names in the matching D3D structs so we don't have to think about this and come up with our own.  Or we need a decoder ring somewhere to help people translate from one naming scheme to another, but that just seems silly to me.